src/io/io.go

```diff
@@ -416,6 +419,10 @@ func copyBuffer(dst Writer, src Reader, buf []byte) (written int64, err error) {
 				err = ew
 				break
 			}
+			if nw < 0 || nw > nr {
+				err = ErrBadWriteCount
+				break
+			}
```

Comment:
This change is not necessary:
1. `nw < 0` is covered by `if ew != nil ` above
2. `nw > nr` is covered by `nr != nw` below. 
3. this new code causes the existing `if nr != nw` logic not reachable _sometimes_
4. why the code do have contain `nw < nr`?